drivers/clock_control/clock_stm32_mco.c

```diff
@@ -0,0 +1,136 @@
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_PLL2CLK
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_PLLI2SCLK
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_PLLI2SCLK
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_PLLI2SCLK_DIV2
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_PLLI2SCLK_DIV2
+#elif DT_PROP(DT_NODELABEL(mco1), source) == STM32_SRC_SYSCLK
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_SYSCLK
+#else
+#error "MCO1 source not a valid clock source"
+#endif
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(mco1), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(mco2), okay)
+#if DT_PROP(DT_NODELABEL(mco2), source) == NO_SEL
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_SYSCLK
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_PLLI2S
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_PLLI2S
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_HSE
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_HSE
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_LSI
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_LSI
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_CSI
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_CSI
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_PLLCLK
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_PLLCLK
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_PLLPCLK
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_PLL1PCLK
+#elif DT_PROP(DT_NODELABEL(mco2), source) == STM32_SRC_PLL2PCLK
+	#define MCO2_SOURCE		LL_RCC_MCO2SOURCE_PLL2PCLK
+#else
+#error "MCO2 source not a valid clock source"
+#endif
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(mco2), okay) */
+
+#define fn_mco_prescaler(i, v) LL_RCC_MCO ## i ## _DIV_ ## v
+#define mco_prescaler(i, v) fn_mco_prescaler(i, v)
+
+#define fn_mco_source(i) MCO ## i ## _SOURCE
+#define mco_source(i) fn_mco_source(i)
+
+struct stm32_mco_config {
+	const struct pinctrl_dev_config *pcfg;
+};
+
+struct stm32_mco_data {
+	uint32_t source;
+	uint32_t prescaler;
+};
+
+static int stm32_mco_init(const struct device *dev)
+{
+	const struct stm32_mco_config *config = dev->config;
+	struct stm32_mco_data *data = dev->data;
+
+	(void)pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	LL_RCC_ConfigMCO(data->source);
+#else
+	LL_RCC_ConfigMCO(data->source, data->prescaler);
+#endif
+
+	return 0;
+}
+
+#define STM32_MCO_INIT(inst)                                            \
+									\
+PINCTRL_DT_INST_DEFINE(inst);                                           \
+									\
+const static struct stm32_mco_config stm32_mco_config_##inst = {        \
+	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(inst),                   \
+};                                                                      \
+static struct stm32_mco_data stm32_mco_data_##inst = {                  \
+	.source = mco_source(DT_INST_REG_ADDR(inst)),                   \
+	.prescaler = mco_prescaler(DT_INST_REG_ADDR(inst),              \
+				   DT_INST_PROP(inst, prescaler)),      \
```

Comment:
Nit 1: could you integrate the `DT_INST_` crust in the `mco_xxx` macros? I will add a suggestion on them.
```suggestion
	.source = mco_source(inst),                   \
	.prescaler = mco_prescaler(inst),             \
```